marlowe-cli/src/Language/Marlowe/CLI/Test/Script.hs

```diff
@@ -96,7 +154,7 @@ interpret Initialize {..} = do
 
 interpret Prepare {..} = do
   marloweTransaction <- findMarloweTransaction soTransaction
-  preparedMarloweTransaction <- prepareTransactionImpl
+  preparedMarloweTransaction <- withError (\(CliError originalMessage) -> CliError $ originalMessage <> " - from Prepare Impl") $ prepareTransactionImpl
```

Comment:
Maybe we can extract this lambda `\(CliError originalMessage) ->...` into a separate local function like `annotateError`. It could be even module level helper like:

```purescript
anotateCliError anotation (CliError msg) = CliError $ annotation <> cliError
```

locally we can have

```purescript
asPrepareError = annotateError "Prepare: "
```